packages/cli/src/util/compile-vercel-config.ts

```diff
@@ -0,0 +1,224 @@
+
+  // Check for conflicting vercel.json and now.json
+  if (hasVercelJson && hasNowJson) {
+    throw new ConflictingConfigFiles([vercelJsonPath, nowJsonPath]);
+  }
+
+  // Only check for vercel.{ext} if feature flag is enabled
+  if (!process.env.VERCEL_TS_CONFIG_ENABLED) {
+    return {
+      configPath: hasVercelJson
+        ? vercelJsonPath
+        : hasNowJson
+          ? nowJsonPath
+          : null,
+      wasCompiled: false,
+    };
+  }
+
+  const vercelConfigPath = await findVercelConfigFile(workPath);
+  const vercelDir = join(workPath, VERCEL_DIR);
+  const compiledConfigPath = join(vercelDir, 'vercel.json');
+
+  if (vercelConfigPath && hasNowJson) {
+    throw new ConflictingConfigFiles(
+      [vercelConfigPath, nowJsonPath],
+      `Both ${basename(vercelConfigPath)} and now.json exist in your project. Please use only one configuration method.`,
+      'https://vercel.com/docs/projects/project-configuration'
+    );
+  }
+
+  if (vercelConfigPath && hasVercelJson) {
+    throw new ConflictingConfigFiles(
+      [vercelConfigPath, vercelJsonPath],
+      `Both ${basename(vercelConfigPath)} and vercel.json exist in your project. Please use only one configuration method.`,
+      'https://vercel.com/docs/projects/project-configuration'
+    );
+  }
+
+  if (!vercelConfigPath) {
+    return {
+      configPath: hasVercelJson
+        ? vercelJsonPath
+        : hasNowJson
+          ? nowJsonPath
+          : null,
+      wasCompiled: false,
+    };
+  }
+
+  dotenvConfig({ path: join(workPath, '.env') });
+  dotenvConfig({ path: join(workPath, '.env.local') });
+
+  const tempOutPath = join(vercelDir, 'vercel-temp.mjs');
+  const loaderPath = join(vercelDir, 'vercel-loader.mjs');
+
+  try {
+    const { build } = await import('esbuild');
+
+    await mkdir(vercelDir, { recursive: true });
+
+    await build({
+      entryPoints: [vercelConfigPath],
+      bundle: true,
+      platform: 'node',
+      format: 'esm',
+      outfile: tempOutPath,
+      packages: 'external',
+      target: 'node20',
+      sourcemap: 'inline',
+    });
+
+    const loaderScript = `
+      import { pathToFileURL } from 'url';
+      const configModule = await import(pathToFileURL(process.argv[2]).href);
+      const config = ('default' in configModule) ? configModule.default : ('config' in configModule) ? configModule.config : configModule;
+      process.send(config);
+    `;
+    await writeFile(loaderPath, loaderScript, 'utf-8');
+
+    const config = await new Promise((resolve, reject) => {
+      const child = fork(loaderPath, [tempOutPath], {
+        stdio: ['pipe', 'pipe', 'pipe', 'ipc'],
+      });
+
+      child.on('message', message => {
+        resolve(message);
+      });
+
+      child.on('error', err => {
+        reject(err);
+      });
+
+      child.on('exit', code => {
+        if (code !== 0) {
+          reject(new Error(`Config loader exited with code ${code}`));
+        }
+      });
```

Comment:
It might be good to also just issue a timeout where we only wait for a message for a couple of seconds. I'm not sure what kinds of things could go wrong, but might as well be safe.